### PR TITLE
Suggest cli based updater in case the instance is bigger 

### PR DESCRIPTION
**Pull request by DeepDiver1975**
refs #23913

@karlitschek 

@jancborchardt UX please

bildschirmfoto von 2016-04-12 16-08-49

### CODE REVIEW

**Comment by karlitschek** [comment]
works for me. but can we do 50 users?
:+1:

**Comment by Xenopathic** [comment]
Please make this setting configurable, not just based on some arbitrary number of users. I want to be able to disable web-based updating on my personal instance, which has <10 users...

File: lib/base.php [hunk 1]
```diff
@@ -361,27 +361,42 @@ public static function checkUpgrade($showTemplate = true) {
...
+		$installedVersion = $systemConfig->getValue('version', '0.0.0');
+		$currentVersion = implode('.', \OCP\Util::getVersion());
+
+		// if not a core upgrade, then it's apps upgrade
+		$isAppsOnlyUpgrade = (version_compare($currentVersion, $installedVersion, '='));
+		if (!$isAppsOnlyUpgrade) {
+			// count users
+			$stats = \OC::$server->getUserManager()->countUsers();
+			$totalUsers = array_sum($stats);
+			if ($totalUsers > 10) {
```
Comment (nickvergessen) [inline lib/base.php]: issue says 100 not 10

**Comment by DeepDiver1975** [comment]
> Please make this setting configurable,

works for me - @karlitschek objections

**Comment by karlitschek** [comment]
hmmm. not sure why someone should configure this. if the admin understands this setting then the same admin can choose the right way to update owncloud in the first place. 
but if there are strong opinions then let's do a config option.

**Comment by Xenopathic** [comment]
@karlitschek For me, it's not about telling the admin to perform the update via occ, but rather *preventing* a random user from performing the update through the web interface, before backups are made etc.

**Comment by DeepDiver1975** [comment]
> hmmm. not sure why someone should configure this. if the admin understands this setting then the same admin can choose the right way to update owncloud in the first place.

thinking about the updater page will popup on any user and any random user might unintentionally trigger the upgrade and the admin having no way to interact - this seems legitimate

**Comment by karlitschek** [comment]
O.K. Let's do a general 'disable web updater' config option instead.
Making the user limit configurable makes no sense.

O.K. for everyone?

**Comment by DeepDiver1975** [comment]
> O.K. Let's do a general 'disable web updater' config option instead.

default being enabled I assume :wink:

**Comment by Xenopathic** [comment]
Until recently there was this other PR to implement exactly that... #18782 :see_no_evil:

**Comment by jancborchardt** [comment]
Only 3 design inputs:
- Typo in header: needs _to_ be updated
- Remove the extra box around the text below the heading. It only needs to be one p element (no class)
- The text should be shortened: Please use the command line updater. For help, see the documentation. (Word »documentation« is the link.)

Ok?

**Comment by Xenopathic** [comment]
You'll want to check the config value in core/ajax/update.php too, to prevent a malicious user from triggering the update (even if the template page says they can't) and potentially causing problems.

**Comment by VicDeo** [comment]
- [ ] in case of the app (not core) upgrade title is still `ownCloud needs be updated to version 9.1.0 pre alpha`

**Comment by DeepDiver1975** [comment]
I suggest to not differentiate between core and app update in this case but to just say that updates need to be installed - any further detailed information is in the cli update command.

**Comment by DeepDiver1975** [comment]
bildschirmfoto von 2016-04-18 10-22-41

@karlitschek @jancborchardt

**Comment by karlitschek** [comment]
nice. Can we add the reason too? Like: 'because it is disabled in the config.php' or 'because you have a big instance with more then xx users'
Beside that. Great 👍

**Comment by jancborchardt** [comment]
Yeah, good call @karlitschek. A reason is good, wording should be like this:

> Please use the command line updater because automatic updating is disabled in the config.php. For help, see the documentation.

> Please use the command line updater because you have a big instance. For help, see the documentation.

Also, the header could be simplified to »Update needed« (no dot at the end).

**Comment by DeepDiver1975** [comment]
bildschirmfoto von 2016-04-18 16-55-06
bildschirmfoto von 2016-04-18 16-55-38


@karlitschek @jancborchardt

**Comment by jancborchardt** [comment]
:+1: nice!

**Comment by karlitschek** [comment]
👍

File: core/ajax/update.php [hunk 2]
```diff
@@ -37,9 +37,17 @@
 // need to send an initial message to force-init the event source,
 // which will then trigger its own CSRF check and produces its own CSRF error
 // message
-$eventSource->send('success', (string)$l->t('Preparing update'));
+//$eventSource->send('success', (string)$l->t('Preparing update'));
```
Comment (DeepDiver1975) [inline core/ajax/update.php]: Note to self: WTF

File: core/ajax/update.php [hunk 3]
```diff
@@ -37,9 +37,17 @@
...
 
 if (OC::checkUpgrade(false)) {
+
+	$config = \OC::$server->getSystemConfig();
+	if ($config->getValue('upgrade.disable-web', true)) {
```
Comment (DeepDiver1975) [inline core/ajax/update.php]: Note to self: WTF

File: core/ajax/update.php [hunk 4]
```diff
@@ -37,9 +37,17 @@
...
+		$eventSource->send('failure', (string)$l->t('Updates need to be installed. Please use the command line updater.'));
```
Comment (MorrisJobke) [inline core/ajax/update.php]: @DeepDiver1975 Also make the message more clear: "Web based update is disabled. Please use the command line tool to upgrade" ;)

Comment (Xenopathic) [inline core/ajax/update.php]: This only gets displayed if a malicious user tries to trigger the update, even though the web UI already says it must be done via occ. Doesn't need to be pretty IMHO

Comment (MorrisJobke) [inline core/ajax/update.php]: > This only gets displayed if a malicious user tries to trigger the update, even though the web UI already says it must be done via occ. Doesn't need to be pretty IMHO

I was hugely confused, what has gone wrong. Better be more clear than to throw badly worded messages.

**Pull request merged**